internal/pkg/deploy/cloudformation/stack/transformers.go

```diff
@@ -116,6 +116,30 @@ func convertContainerHealthCheck(hc manifest.ContainerHealthCheck) *template.Con
 	}
 }
 
+func convertHostedZone(m manifest.RoutingRuleConfiguration) template.HostedZoneToAliases {
+	hostedZoneToAliases := make(map[string][]string)
+	defaultHostedZone := m.HostedZone
+	for _, alias := range m.Alias.AdvancedAliases {
+		if alias.HostedZone != nil {
+			hostedZoneToAliases[*alias.HostedZone] = append(hostedZoneToAliases[*alias.HostedZone], *alias.Alias)
+			continue
+		}
+		if defaultHostedZone != nil {
+			hostedZoneToAliases[*defaultHostedZone] = append(hostedZoneToAliases[*defaultHostedZone], *alias.Alias)
+		}
+	}
+	if defaultHostedZone == nil {
+		return hostedZoneToAliases
+	}
+	for _, alias := range m.Alias.StringSliceOrString.StringSlice {
+		hostedZoneToAliases[*defaultHostedZone] = append(hostedZoneToAliases[*defaultHostedZone], alias)
+	}
+	if m.Alias.StringSliceOrString.String != nil {
+		hostedZoneToAliases[*defaultHostedZone] = append(hostedZoneToAliases[*defaultHostedZone], *m.Alias.StringSliceOrString.String)
+	}
```

Comment:
Do you think we can use `m.Alias.ToStringSlice` here to simplify this a little bit?